File: cmi-api/src/main/java/fr/cmi/member/domain/Member.java
package fr.cmi.member.domain;

import lombok.*;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@EqualsAndHashCode
@Getter
@Builder
@ToString
@AllArgsConstructor
public class Member {

    @NotNull
    private String id;

    @NotNull
    private String firstName;

    @NotNull
    private String lastName;

    @NotNull
    private Long age;

    @NotNull
    private LocalDate birthDay;

    @NotNull
    private Role role;

    @NotNull
    private Gender gender;

    @NotNull
    private Status status;

    private String classe;
}

File: cmi-app/src/app/members/components/member-list/member.ts
import {Role} from "./role";

export class Member {

  firstName: string;
  lastName: string;
  classe: string;
  age: number;
  role: Role;

  constructor(firstName: string, lastName: string, classe: string, age: number, role: Role) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.classe = classe;
    this.age = age;
    this.role = role;
  }
}
